pandas/core/arrays/base.py

```diff
@@ -610,3 +615,118 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+
+class ExtensionOpsMixin(object):
+    """
+    A base class for linking the operators to their dunder names
+    """
+    @classmethod
+    def addArithmeticOps(cls):
+        cls.__add__ = cls._create_method(operator.add)
+        cls.__radd__ = cls._create_method(ops.radd)
+        cls.__sub__ = cls._create_method(operator.sub)
+        cls.__rsub__ = cls._create_method(ops.rsub)
+        cls.__mul__ = cls._create_method(operator.mul)
+        cls.__rmul__ = cls._create_method(ops.rmul)
+        cls.__pow__ = cls._create_method(operator.pow)
+        cls.__rpow__ = cls._create_method(ops.rpow)
+        cls.__mod__ = cls._create_method(operator.mod)
+        cls.__rmod__ = cls._create_method(ops.rmod)
+        cls.__floordiv__ = cls._create_method(operator.floordiv)
+        cls.__rfloordiv__ = cls._create_method(ops.rfloordiv)
+        cls.__truediv__ = cls._create_method(operator.truediv)
+        cls.__rtruediv__ = cls._create_method(ops.rtruediv)
+        if not PY3:
+            cls.__div__ = cls._create_method(operator.div)
+            cls.__rdiv__ = cls._create_method(ops.rdiv)
+    
+        cls.__divmod__ = cls._create_method(divmod)
+        cls.__rdivmod__ = cls._create_method(ops.rdivmod)
+        
+    @classmethod
+    def addComparisonOps(cls):
+        cls.__eq__ = cls._create_method(operator.eq, False)
+        cls.__ne__ = cls._create_method(operator.ne, False)
+        cls.__lt__ = cls._create_method(operator.lt, False)
+        cls.__gt__ = cls._create_method(operator.gt, False)
+        cls.__le__ = cls._create_method(operator.le, False)
+        cls.__ge__ = cls._create_method(operator.ge, False)
```

Comment:
The `False` here is quite specific to the `_create_method` implementation of the ScalarOpsMixin, so this doesn't feel completely "clean". 

So you could also have a `_create_arithmetic_method` and `_create_comparison_method` class methods, and in your scalar mixing below you can reuse the same function for both those class methods (with a different value of the keyword).